test/controller_SUITE.erl

```diff
@@ -0,0 +1,181 @@
+     add_bad_range_of_users_to_started_scenario_fails,
+     add_users_scenario_not_started_fails,
+     remove_users_scenario_not_started_fails,
+     check_status_with_running_users_is_correct,
+     check_status_after_killing_one_user_is_correct,
+     stop_non_running_scenario_fails,
+     stop_running_scenario_with_no_users_immediately_terminates,
+     stop_running_scenario_with_users_stays_in_finished,
+     stop_running_scenario_with_users_eventually_terminates,
+     scenario_with_state_and_crashing_in_terminate_run_fine
+    ].
+
+
+%% setup and teardown
+init_per_suite(Config) ->
+    Config.
+
+end_per_suite(Config) ->
+    Config.
+
+init_per_testcase(_TestCase, Config) ->
+    application:ensure_all_started(amoc),
+    Config.
+
+end_per_testcase(_TestCase, Config) ->
+    application:stop(amoc),
+    Config.
+
+%% test cases
+no_scenario_running_status_is_idle(_) ->
+    Status = amoc_controller:get_status(),
+    ?assertMatch(idle, Status).
+
+disable_controller_returns_status_disable(_) ->
+    amoc_controller:disable(),
+    Status = amoc_controller:get_status(),
+    ?assertMatch(disabled, Status).
+
+disable_controller_then_cant_start_scenario(_) ->
+    amoc_controller:disable(),
+    Ret = do_start_scenario(testing_scenario),
+    ?assertMatch({error, {invalid_status, disabled}}, Ret).
+
+start_non_existing_scenario_fails(_) ->
+    Scenario = 'non_existing_come_on_dont_do_this_#(*&$%*(@_))',
+    Ret = do_start_scenario(Scenario),
+    ?assertMatch({error, {no_such_scenario, Scenario}}, Ret).
+
+start_scenario_without_vars_fails(_) ->
+    Ret = do_start_scenario(testing_scenario),
+    ?assertMatch({error, {parameters_verification_failed, [{testing_var1, _, _}]}}, Ret).
+
+start_scenario_runs_fine(_) ->
+    Ret = do_start_scenario(testing_scenario, regular_vars()),
+    ?assertMatch(ok, Ret).
+
+start_scenario_check_status(_) ->
+    do_start_scenario(testing_scenario, regular_vars()),
+    Status = amoc_controller:get_status(),
+    ?assertMatch({running, testing_scenario, 0, 0}, Status).
+
+add_users_to_started_scenario(_) ->
+    do_start_scenario(testing_scenario, regular_vars()),
+    Ret = amoc_controller:add_users(1, 10),
+    ?assertMatch(ok, Ret).
+
+add_bad_range_of_users_to_started_scenario_fails(_) ->
+    do_start_scenario(testing_scenario, regular_vars()),
+    Ret = amoc_controller:add_users(10, 1),
+    ?assertMatch({error, invalid_range}, Ret).
+
+add_users_scenario_not_started_fails(_) ->
+    Ret = amoc_controller:add_users(1, 10),
+    ?assertMatch({error, {invalid_status, idle}}, Ret).
+
+remove_users_scenario_not_started_fails(_) ->
+    Ret = amoc_controller:add_users(1, 10),
+    ?assertMatch({error, {invalid_status, idle}}, Ret).
+
+check_status_with_running_users_is_correct(_) ->
+    do_start_scenario(testing_scenario, regular_vars()),
+    amoc_controller:add_users(1, 10),
+    Status = amoc_controller:get_status(),
+    ?assertMatch({running, testing_scenario, _, 10}, Status).
```

Comment:
think we should check the number of users as well:
```
?assertEqual({running, testing_scenario, 10, 10}, Status)
```
 and it's better to use range starting from some other value than 1:
 ```
    amoc_controller:add_users(6, 10),
    Status = amoc_controller:get_status(),
    timer:sleep(50*6), % enough time to start 6 users with 50ms interarrival
    ?assertEqual({running, testing_scenario, 5, 10}, Status).
 ```